src/shared/contexts/app.tsx

```diff
@@ -159,6 +169,10 @@ export const AppSettingProvider: FC = ({children}) => {
     dispatch(setWorkerRegistrationInfluxQL(workerRegistrationInfluxQL))
   }, [workerRegistrationInfluxQL])
 
+  useEffect(() => {
+    dispatch(setWorkerRegistrationSQL(workerRegistrationSQL))
+  }, [workerRegistrationSQL])
+
```

Comment:
Does this need to be registered multiple times, or is once on app load if sufficient? If the latter, it might make more sense to add an `if remoteDataState != RemoteDataState.Done {}` style check to ensure this useEffect hook is only run once.

Reply:
Good point! Like you mentioned in the latter point, this service worker is registered once before the app load. I like the idea of ensuring the code to run `dispatch(setWorkerRegistrationSQL(workerRegistrationSQL))` once 😃

Since this context provider component runs before the page load, there is no data yet, so we cannot check on `remoteDataState` at this moment. 

One thing I can think of is to register the service worker when the component is just mounted, which only runs once. Something like this:

```js
useEffect(() => {
    dispatch(setWorkerRegistration(workerRegistration))
    dispatch(setWorkerRegistrationInfluxQL(workerRegistrationInfluxQL))
    dispatch(setWorkerRegistrationSQL(workerRegistrationSQL))
  }, [])
```

```js
// useEffect hook with the empty array is equivalent to componentDidMount() and only run once
useEffect(() => {
  ...
}, [])
```
Where `workerRegistration` and `workerRegistrationInfluxQL` are service workers for Flux and InfluxQL. Since each service worker can only serve on HTTP path, that's why we need three different service workers to support `/api/v2private/query` (SQL), `/query` (InfluxQL), and `/api/v2/query` (Flux) separately

What do you think?

Reply:
I don't usually love this approach:

```
useEffect(() => {
    dispatch(setWorkerRegistration(workerRegistration))
    dispatch(setWorkerRegistrationInfluxQL(workerRegistrationInfluxQL))
    dispatch(setWorkerRegistrationSQL(workerRegistrationSQL))
  }, [])
```

for two reasons. For one, it's somewhat anti-pattern in React because the linter will probably warn you that the variables (like dispatch or setWorkerRegistration...) need to be included in the dependency array. The way you did it previously is completely right IMHO, as long as there's not a risk that the equality check fails for `workerRegistrationSQL` and causes extra re-renders. 

I think a way to test this would be -  keep the useEffect hook like this, but test it (for example, temporarily add a `console.log`) to make sure the hook isn't being called multiple times on app load. If it isn't, I actually wouldn't change anything.

```
useEffect(() => {
    dispatch(setWorkerRegistrationSQL(workerRegistrationSQL))
  }, [workerRegistrationSQL])
```

Reply:
I tested using a `console.log()` in the useEffect hook and confirmed that the useEffect hook only run once:

```
useEffect(() => { 
  dispatch(setWorkerRegistrationSQL(workerRegistrationSQL))
}, [workerRegistrationSQL])
```

Then I'll keep the original code. Thanks for the brainstorming 😄🧠